pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1939,22 +1940,35 @@ public TopicStatsImpl getStats(boolean getPreciseBacklog, boolean subscriptionBa
                         ledgers.forEach(ledgerId -> {
                             CompletableFuture<Void> completableFuture = new CompletableFuture<>();
                             getLedgerMetadataFutures.add(completableFuture);
-                            brokerService.getPulsar().getBookKeeperClient()
-                                    .getLedgerMetadata(ledgerId)
-                                    .thenAccept(metadata -> {
-                                        LedgerInfo schemaLedgerInfo = new LedgerInfo();
-                                        schemaLedgerInfo.ledgerId = metadata.getLedgerId();
-                                        schemaLedgerInfo.entries = metadata.getLastEntryId() + 1;
-                                        schemaLedgerInfo.size = metadata.getLength();
-                                        if (includeLedgerMetadata) {
-                                            info.metadata = metadata.toSafeString();
-                                        }
-                                        stats.schemaLedgers.add(schemaLedgerInfo);
-                                        completableFuture.complete(null);
-                                    }).exceptionally(e -> {
-                                completableFuture.completeExceptionally(e);
-                                return null;
-                            });
+                            CompletableFuture<LedgerMetadata> metadataFuture =
+                                    null;
+                            try {
+                                metadataFuture = brokerService.getPulsar().getBookKeeperClient()
+                                .getLedgerMetadata(ledgerId);
+                            } catch (Exception e) {
```

Comment:
Why catch Exception here, can we just use `whenComplete` ?

Reply:
No, the bookkeeper client throws NPE here.

Reply:
can you open a ticket on BK ?

then we should catch only NPE and add a link to the BK issue

Reply:
Done